### PR TITLE
Add mergeable libraries support to dynamic libraries

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Add `make_mergable` linker flags to supported SDKs

From the `ld` manual. Opened in a file so it's easier to browse.

```bash
man ld | col -b | open -tf
```

```
Mergeable Library Options
 -make_mergeable
     Adds additional metadata to a dylib which makes it a mergeable
     library.  It can still be used as a dylib, or can be merged into
     other binaries when they link it with a -merge* option.
```

Also, users have to build with Xcode >= 15.

```
When you include your mergeable library in an XCFramework, Xcode adds the key MergeableMetadata to the XCFramework’s Info.plist file to indicate to other projects that your library in the XCFramework is mergeable. You can only use XCFrameworks with mergeable metadata in Xcode 15 and later; in earlier versions, Xcode returns a build error.
```

## :bulb: Motivation and Context

Users can enable mergable libraries featue for dynamic sentry frameworks.

See https://developer.apple.com/documentation/xcode/configuring-your-project-to-use-mergeable-libraries for more info.

## :green_heart: How did you test it?

Closes https://github.com/getsentry/sentry-cocoa/issues/3719

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Comment by denrase** [comment]
I'm using Xcode 16.0 (16A242d) on macOS 15.0 (24A335) and i'm currently not able to build an app binary which embeds the frameworks into the app binary. The sample from #4301 is also not working anymore. Also tried on another machine with Xcode 15.1 with the same result.

**Comment by denrase** [comment]
Turns out just enabling it is not enough, we also have to remove `Sentry` from the `Embedded Frameworks` in the sample applications. Just setting to automatic will not remove the framework, but leave it there and also embed it in the app.

**Comment by denrase** [comment]
Updated the build script. watchos and watchsimulator cant even build if the MERGED_BINARY_TYPE key is present, even if set to none. So we temporarily remove it from the project file using sed.

Very annoying.

This seems kinday finicky to me and adding too much complexity just to have the samples be mergale binaries. We could revert the samples and the `Sentry.framework` target config from supporting mergeable.

@brustolin @philipphofmann WDYT?

**Comment by brustolin** [comment]
> This seems kinday finicky to me and adding too much complexity just to have the samples be mergale binaries. We could revert the samples and the Sentry.framework target config from supporting mergeable.

Im not sure If I get this. 
We dont need the samples to use mergeable binaries if this makes things harder. But we do need at least one Test project for at least one target (iOS most likely) that have mergeable enabled that uses Sentry.

File: scripts/build-xcframework.sh [hunk 1]
```diff
@@ -16,22 +15,51 @@ generate_xcframework() {
     local configuration_suffix="${4-}"
     local createxcframework="xcodebuild -create-xcframework "
     local GCC_GENERATE_DEBUGGING_SYMBOLS="YES"
-    
+    local OTHER_LDFLAGS="-Wl,-make_mergeable" # Default value with -make_mergeable flag
```
Comment (brustolin) [inline scripts/build-xcframework.sh:18]: l: Do we still need this flags after enabling `MERGEABLE_LIBRARY` in the project settings?

**Comment by denrase** [comment]
Closed in favor of #4381

**Pull request closed without merging**